src/Pkg/Package.hs

```diff
@@ -287,20 +286,24 @@ buildMods opts ns = do let f = map (toPath . showCG) ns
     where toPath n = foldl1' (</>) $ splitOn "." n
 
 testLib :: Bool -> String -> String -> IO Bool
-testLib warn p f
-    = do d <- getDataDir
-         gcc <- getCC
-         (tmpf, tmph) <- tempfile ""
-         hClose tmph
-         let libtest = d </> "rts" </> "libtest.c"
-         e <- rawSystem gcc [libtest, "-l" ++ f, "-o", tmpf]
-         case e of
-            ExitSuccess -> return True
-            _ -> do if warn
-                       then do putStrLn $ "Not building " ++ p ++
-                                          " due to missing library " ++ f
-                               return False
-                       else fail $ "Missing library " ++ f
+testLib warn p f = do
```

Comment:
This is not the removal of an extra "do". Instead, it's a conversion to dangling position, which IMHO makes things much harder to read because the reason for the switch to imperative mode is not where it happens. We haven't used that style historically in most of Idris, and it seems that a global switch to dangling mode should at least merit a bit of discussion.